src/backend/sycl_backend.cc

```diff
@@ -124,34 +145,56 @@ struct sycl_backend::impl {
 
 	thread_queue& get_host_queue(const size_t lane) {
 		assert(lane <= host.queues.size());
-		if(lane == host.queues.size()) { host.queues.emplace_back(fmt::format("cy-host-{}", lane), enable_profiling); }
+		if(lane == host.queues.size()) { host.queues.emplace_back(fmt::format("cy-host-{}", lane), config.profiling); }
 		return host.queues[lane];
 	}
 
 	sycl::queue& get_device_queue(const device_id did, const size_t lane) {
 		auto& device = devices[did];
 		assert(lane <= device.queues.size());
 		if(lane == device.queues.size()) {
-			const auto properties = enable_profiling ? sycl::property_list{sycl::property::queue::enable_profiling{}, sycl::property::queue::in_order{}}
+			const auto properties = config.profiling ? sycl::property_list{sycl::property::queue::enable_profiling{}, sycl::property::queue::in_order{}}
 			                                         : sycl::property_list{sycl::property::queue::in_order{}};
 			device.queues.emplace_back(device.sycl_device, sycl::async_handler(sycl_backend_detail::report_errors), properties);
 		}
 		return device.queues[lane];
 	}
 };
 
-sycl_backend::sycl_backend(const std::vector<sycl::device>& devices, const bool enable_profiling) : m_impl(new impl(devices, enable_profiling)) {}
+sycl_backend::sycl_backend(const std::vector<sycl::device>& devices, const configuration& config) : m_impl(new impl(devices, config)) {
+	// Initialize a submission thread with hydrator for each device, if they are enabled
+	if(m_impl->config.per_device_submission_threads) {
+		for(device_id did = 0; did < m_impl->system.devices.size(); ++did) {
+			m_impl->devices[did].submission_thread.emplace(fmt::format("cy-be-submission-{}", did.value), m_impl->config.profiling);
+			// no need to wait for the event -> will happen before the first task is submitted
+			(void)m_impl->devices[did].submission_thread->submit([did] {
+				closure_hydrator::make_available();
+				// TODO: Set BE submission thread affinity
+			});
+		}
+	}
```

Comment:
```suggestion
	} else {
	    closure_hydrator::make_available();
	}
```

Could we move this call from the `live_executor` to here? If submission threads are enabled then we don't need it to make it available it in the executor. Besides, it's somewhat confusing to have it called in two places.

Reply:
As discussed offline, still required in some testing / non-live-execution scenarios. Documented.